ca_gazebo/include/ca_gazebo/create_cliff_plugin.h

```diff
@@ -0,0 +1,78 @@
+#ifndef GAZEBO_ROS_CLIFF_SENSOR_HH
+#define GAZEBO_ROS_CLIFF_SENSOR_HH
+
+#include <ros/ros.h>
+#include <ros/advertise_options.h>
+#include <tf/tf.h>
+#include <tf/transform_listener.h>
+
+#include <vector> 
+
+#include <gazebo/transport/transport.hh>
+#include <gazebo/plugins/RayPlugin.hh>
+
+#include <gazebo_plugins/gazebo_ros_utils.h>
+#include <gazebo_plugins/PubQueue.h>
+#include <std_msgs/Bool.h>
+
+namespace gazebo
+{
+  class GazeboRosCliff : public RayPlugin
+  {
+    /// \brief Constructor
+    public: GazeboRosCliff();
+
+    /// \brief Destructor
+    public: ~GazeboRosCliff();
+
+    /// \brief Load the plugin
+    /// \param take in SDF root element
+    public: void Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf);
+
+    /// \brief Keep track of number of connctions
+    private: int laser_connect_count_;
+    private: void LaserConnect();
+    private: void LaserDisconnect();
+
+    // Pointer to the model
+    GazeboRosPtr gazebo_ros_;
+    private: std::string world_name_;
+    private: physics::WorldPtr world_;
+    /// \brief The parent sensor
+    private: sensors::RaySensorPtr parent_ray_sensor_;
+
+    /// \brief pointer to ros node
+    private: ros::NodeHandle* rosnode_;
```

Comment:
You don't need `rosnode_` since you have a `GazeboRosPtr` instance. Check [this](https://github.com/ros-simulation/gazebo_ros_pkgs/blob/311c56637825c64c9e330a93ea3fc1e305daf435/gazebo_plugins/src/gazebo_ros_utils.cpp#L45).

Reply:
I'm also not sure how to deal with deas, should I create that Ptr in my page, or should I use it, because I don't know how